File: Test/regression/C/Sterbenz.c
#include <stdio.h>
#include <float.h>
double av1(double x, double y){
	double z;
	z = (x+y)/2.0;
	return z;
}
double av2(double x, double y){
	double z;	
	z = x/2.0 + y/2.0;
	return z;
}
double av3(double x,double y){
	double z;
	z = x + (y-x)/2.0;
	return z;
}
double av4(double x, double y){
	double z;
	z = y + (x-y)/2.0;
	return z;
}
double average(double x, double y){
	int samesign;
	double avg;
	if ( x >= 0 ){
		if (y >=0)
			samesign = 1;
		else
			samesign = 0;	
	}else{
		if (y >= 0)
			samesign = 0;
		else
			samesign = 1;	
	}
	if (samesign){
		if ( y >= x)
			avg = av3(x,y);
		else
			avg = av4(x,y);	
	}else
		avg = av1(x,y);
	return avg;	
}

File: Test/regression/Fortran/sterbenz.f90
DOUBLE PRECISION function average (x, y)
DOUBLE PRECISION x, y, zero, two, av1, av2, av3, av4 
logical samesign
parameter (zero = 0.0e+00, two = 2.0e+00)
av1(x,y) = (x + y) / two
av2(x,y) = (x / two) + (y / two)
av3(x,y) = x + ((y - x) / two)
av4(x,y) = y + ((x - y) / two)

if (x .ge. zero) then
if (y .ge. zero) then
samesign = .true.
else
samesign = .false.
endif
else
if (y .ge. zero) then
samesign = .false.
else
samesign = .true.
endif
endif

if (samesign) then
if (y .ge. x) then
average = av3(x,y)
else
average = av4(x,y)
endif
else
average = av1(x,y)
endif

return
end